kernel/preemption/src/lib.rs

```diff
@@ -3,30 +3,14 @@
 //! Supports enabling and disabling preemption for the purpose of 
 //! safe task state management, e.g., through preemption-safe locks.
 
-use core::sync::atomic::{AtomicU8, Ordering};
-use cpu::CpuId;
-use crate::cpu_local::{CpuLocal, CpuLocalField, PerCpuField};
-use no_drop::NoDrop;
+#![no_std]
+#![feature(negative_impls, thread_local)]
 
-/// The preemption count for the current CPU.
-static PREEMPTION_COUNT: CpuLocal<PreemptionCount> = CpuLocal::new(PerCpuField::PreemptionCount);
+use cpu::CpuId;
 
-/// A type wrapper around [`AtomicU8`] that represents a CPU-local preemption count.
-///
-/// If a CPU's preemption count is `0`, preemption is enabled.
-/// If a CPU's preemption count is greater than `0`, preemption is disabled.
-pub struct PreemptionCount(AtomicU8);
-impl PreemptionCount {
-    /// Creates a new `PreemptionCount` with an initial value of `0`.
-    pub const fn new() -> Self {
-        PreemptionCount(AtomicU8::new(0))
-    }
-}
-// SAFETY: The `PreemptionCount` type corresponds to a field in `PerCpuData`
-//         with the offset specified by `PerCpuField::PreemptionCount`.
-unsafe impl CpuLocalField for PreemptionCount {
-    const FIELD: PerCpuField = PerCpuField::PreemptionCount;
-}
+// NOTE: This offset must be kept in sync with `cpu_local::PerCpuField`.
+#[cls::cpu_local(12)]
+static PREEMPTION_COUNT: u8 = 0;
```

Comment:
this uses a `u8` here but in the `PerCpuData` struct you use a `u16`. I don't particularly mind which one we pick, but let's keep it consistent.